Add unit tests

theendlessriver13 opened the pull request:
Hi,
I started implementing some unit tests for this module. I hope this is okay...
If I find any bugs during this process, I will send a separate patch and rebase this branch onto the main branch.

As you can see, this is just a draft and still work in progress. I am aiming for full coverage which may take a while.

I am unsure how your licensing conventions work, please let me know if there is something wrong.

tannewt commented:
Looks great! Thanks for checking in. I'd love to merge these in. Let's make sure they run in GitHub Actions too.

theendlessriver13 commented:
thanks, I added a few more tests and also modified gha to run the tests, which seems to work, however I am not sure if this is the way you like them to run and if we want to to tests against different version of (C)Python ?!

tannewt commented:
Looks all good to me! I think you have a better understanding of best test practices than I do. Thanks!
